### ISSUE TITLE
Update Utf8JsonWriter.WriteNumber and WriteNumberValue(double/float) to support infinity correctly for all cases

### ISSUE THREAD

**Issue by ahsonkhan** [dotnet/corefx#40133]
This PR https://github.com/dotnet/corefx/pull/38322 attempted to fix https://github.com/dotnet/corefx/issues/33997 but did not update all cases.

For example, the fix was made for `JsonWriterOptions.Indented = true`, but not for false when writing a double.
For example, this needs to be fixed (in `WriteNumberValueMinimized`):
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs#L59-L61

Similarly here:
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Float.cs#L59-L61

Here's an example of the fix (in `WriteNumberValueIndented`):
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs#L93-L95
Calling:
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs#L98

Also add tests to make sure all the scenarios are covered (all permutation of `JsonWriterOptions`).

Similarly, all the `WriteNumber(some property name string, double/float value)` API overloads need to be fixed as well:
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Float.cs#L242-L244

cc @tannergooding

**Comment by WinCPP** [comment]
@ahsonkhan I submitted the changes to use `TryFormatDouble` or `TryFormatFloat` respectively in the necessary places. Not sure if that was all to be done here.

> Also add tests to make sure all the scenarios are covered (all permutation of JsonWriterOptions).

Looks like all combinations of `Indented` and `SkipValidation` of `JsonWriterOptions` have already been handled in following places. I didn't get which specific tests scenarios we would want to add further. Appreciate inputs. Thanks!

For double: https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/tests/Utf8JsonWriterTests.cs#L4934-L4939

For float:
https://github.com/dotnet/corefx/blob/902dff43381cd4ce6bf1dec01618212a5f8f0231/src/System.Text.Json/tests/Utf8JsonWriterTests.cs#L4906-L4911

**Comment by ahsonkhan** [comment]
The value being tested in the existing test (`1234567`) doesn't have any roundtripping issues which is why it works fine.

Add a test case where double/float doesn't roundtrip. On .NET core 3.0+, everything works as expected. But on older frameworks (including netfx), certain double/float values wouldn't roundtrip unless G17/G9 was used.

So for example, try `double.MaxValue`/`double.MinValue`.

On netcoreapp: It will work
On netfx: It won't (particularly when formatted = false because the fix was already made for formatted = true)

Similarly, if you test the `WriteString("propertyName", double.MaxValue)`.

Same with single (`float.MaxValue`, `float.MinValue`).

```text
System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(value: 1.79769313486232E+308, formatted: False, skipValidation: True) [FAIL]
        Assert.Equal() Failure
                                   (pos 16)
        Expected: [1.7976931348623157E+308,1.7976931348623157E+308,1.797693···
        Actual:   [1.79769313486232E+308,1.79769313486232E+308,1.7976931348···
                                   (pos 16)
        Stack Trace:
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\JsonTestHelper.cs(693,0): at System.Text.Json.JsonTestHelper.AssertContents(String expectedValue, ArrayBufferWriter`1 buffer)
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(5000,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(Double value, Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(value: 1.79769313486232E+308, formatted: False, skipValidation: False) [FAIL]
        Assert.Equal() Failure
                                   (pos 16)
        Expected: [1.7976931348623157E+308,1.7976931348623157E+308,1.797693···
        Actual:   [1.79769313486232E+308,1.79769313486232E+308,1.7976931348···
                                   (pos 16)
        Stack Trace:
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\JsonTestHelper.cs(693,0): at System.Text.Json.JsonTestHelper.AssertContents(String expectedValue, ArrayBufferWriter`1 buffer)
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(5000,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(Double value, Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(value: -1.79769313486232E+308, formatted: False, skipValidation: True) [FAIL]
        Assert.Equal() Failure
                                    (pos 17)
        Expected: [-1.7976931348623157E+308,-1.7976931348623157E+308,-1.7976···
        Actual:   [-1.79769313486232E+308,-1.79769313486232E+308,-1.79769313···
                                    (pos 17)
        Stack Trace:
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\JsonTestHelper.cs(693,0): at System.Text.Json.JsonTestHelper.AssertContents(String expectedValue, ArrayBufferWriter`1 buffer)
          E:\GitHub\Fork\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(5000,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(Double value, Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteNumberValueDouble(value: -1.79769313486232E+308, formatted: False, skipValidation: False) [FAIL]
        Assert.Equal() Failure
                                    (pos 17)
        Expected: [-1.7976931348623157E+308,-1.7976931348623157E+308,-1.7976···
        Actual:   [-1.79769313486232E+308,-1.79769313486232E+308,-1.79769313···
                                    (pos 17)
```

**Comment by WinCPP** [comment]
@ahsonkhan Thanks for inputs. Now I understand. I got fixated with 'all permutations of JsonWriterOptions' :-)

**Comment by ahsonkhan** [comment]
Re-opening until its fixed in release/3.0.

**Comment by ahsonkhan** [comment]
Fixed in 3.0 by https://github.com/dotnet/corefx/pull/40419
